### PR TITLE
Kill ServoParser::pipeline

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15220)

### CODE REVIEW

**Comment by jdm** [comment]
@bors-servo: retry
* I suspect builder shenanigans

**Comment by nox** [comment]
@jdm Seems like this explicit reflow might be important. :P

**Comment by nox** [comment]
I moved the explicit reflow in the `PageSource` arm in `finish_load`.

@bors-servo r=SimonSapin,jdm

File: components/script/dom/document.rs [hunk 1]
```diff
@@ -1556,6 +1551,15 @@ impl Document {
                 self.process_deferred_scripts();
             },
             LoadType::PageSource(_) => {
+                if self.browsing_context.is_some() {
+                    // Disarm the reflow timer and trigger the initial reflow.
+                    self.reflow_timeout.set(None);
+                    self.upcast::<Node>().dirty(NodeDamage::OtherNodeDamage);
+                    self.window.reflow(ReflowGoal::ForDisplay,
+                                    ReflowQueryType::NoQuery,
```
Comment (Ms2ger) [inline components/script/dom/document.rs]: Indentation?

Comment (nox) [inline components/script/dom/document.rs]: I swear this looked okay in my editor. I'm so bad at this.

**Pull request merged**